presto-main/src/main/java/com/facebook/presto/operator/scalar/FormatUtilsFunctions.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.StandardTypes;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+
+import static com.facebook.presto.spi.StandardErrorCode.NUMERIC_VALUE_OUT_OF_RANGE;
+import static com.facebook.presto.util.Failures.checkCondition;
+import static java.lang.Float.intBitsToFloat;
+import static java.lang.String.format;
+
+public final class FormatUtilsFunctions
+{
+    private FormatUtilsFunctions() {}
+
+    @Description("format to human readable count value")
+    @ScalarFunction("format_count")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice formatCountTinyint(@SqlType(StandardTypes.TINYINT) long num)
+    {
+        checkCondition(num != Byte.MIN_VALUE, NUMERIC_VALUE_OUT_OF_RANGE, "Value -128 is out of range for abs(tinyint)");
+        return Slices.utf8Slice(fmtCount(num));
+    }
+
+    @Description("format to human readable count value")
+    @ScalarFunction("format_count")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice formatCountSmallint(@SqlType(StandardTypes.SMALLINT) long num)
+    {
+        checkCondition(num != Short.MIN_VALUE, NUMERIC_VALUE_OUT_OF_RANGE, "Value -32768 is out of range for abs(smallint)");
+        return Slices.utf8Slice(fmtCount(num));
+    }
+
+    @Description("format to human readable count value")
+    @ScalarFunction("format_count")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice formatCountInteger(@SqlType(StandardTypes.INTEGER) long num)
+    {
+        checkCondition(num != Integer.MIN_VALUE, NUMERIC_VALUE_OUT_OF_RANGE, "Value -2147483648 is out of range for abs(integer)");
+        return Slices.utf8Slice(fmtCount(num));
+    }
+
+    @Description("format to human readable count value")
+    @ScalarFunction("format_count")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice formatCount(@SqlType(StandardTypes.BIGINT) long num)
```

Comment:
Not necessarily.
Please have a look on an example of float and double. Despite the fact float -> double coercion exists and we could compute double functions on float input, the result would be misleading.
E.g.:
`select sum(cast(x as double)) from (values (real '1.23')) t(x);` which emulates not having real-specific `sum` function.
returns `1.2300000190734863` instead of `1.23` (as float AKA real).